textselect.cpp

```diff
@@ -16,6 +16,17 @@
 
 #include "textselect.hpp"
 
+// Calculates the midpoint between two numbers
+template<typename T>
+constexpr T midpoint(T a, T b) {
+    return a + (b - a) / 2;
+}
+
+// Checks if a string view ends with the specified char suffix
+bool ends_with(std::string_view str, char suffix) {
```

Comment:
```suggestion
bool endsWith(std::string_view str, char suffix) {
```

This should be named `endsWith` to match the camelCase style used in the rest of the code.